nodejs/eks/nodes/instances.ts

```diff
@@ -0,0 +1,209 @@
+ * - Interface type "efa"
+ *
+ * Any secondary network interfaces are configured with:
+ * - No public IP
+ * - Device index 1
+ * - Interface type "efa-only"
+ *
+ * @param args - Configuration options for the network interfaces
+ * @returns Array of network interface configurations for the launch template
+ * @throws {pulumi.InputPropertyError} If the selected instance type does not support EFA
+ */
+export function getEfaNetworkInterfaces(
+    args: GetEfaNetworkInterfacesArgs,
+): pulumi.Output<aws.types.input.ec2.LaunchTemplateNetworkInterface[]> {
+    const instanceType = getEfaInstanceType(args.instanceTypes);
+    const instanceTypeInfo = aws.ec2.getInstanceTypeOutput(
+        {
+            instanceType,
+        },
+        args.opts,
+    );
+
+    // Instance types that support multiple network cards can be configured with one EFA per network card. All other supported instance types
+    // support only one EFA per instance.
+    // See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-limits
+    const maximumNetworkCards = pulumi
+        .all([instanceTypeInfo.efaSupported, instanceType])
+        .apply(([efaSupported, instanceType]) => {
+            if (!efaSupported) {
+                throw new pulumi.InputPropertyError({
+                    propertyPath: args.instanceTypePropertyPath,
+                    reason: `The selected instance type '${instanceType}' does not support EFA.`,
+                });
+            }
+            return instanceTypeInfo.maximumNetworkCards.apply((cards) => {
+                return cards > 1 ? cards : 1;
+            });
+        });
+
+    const publicIpAssociation = args.associatePublicIpAddress
+        ? pulumi.output(args.associatePublicIpAddress).apply((str) => str.toString())
+        : undefined;
+
+    return maximumNetworkCards.apply((cards) => {
+        const interfaces: aws.types.input.ec2.LaunchTemplateNetworkInterface[] = [];
+        for (let i = 0; i < cards; i++) {
+            interfaces.push({
+                // The primary interface can have a public IP address, the secondary interfaces do not support IP networking.
+                associatePublicIpAddress: i === 0 ? publicIpAssociation : undefined,
+                deleteOnTermination: "true",
+                networkCardIndex: i,
+                // EFA requires the first interface to have device index 0, and all the secondary interfaces to have device index 1.
+                // see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-tempinstance
+                deviceIndex: i === 0 ? 0 : 1,
+                // The primary interface must be efa, the secondary interfaces can be efa or efa-only.
+                // We choose efa-only because it prevents exhausting the private IP address space and circumvents
+                // possibly routing conflicts, like source IP mismatch, or host name mapping problems.
+                interfaceType: i === 0 ? "efa" : "efa-only",
```

Comment:
Making this configurable would yield a rather clunky API. A list of enums?
This would require users to know how many cards their instance type has.

In this case it's better if they pass a custom launch template instead.